src/shogun/distance/MahalanobisDistance.cpp

```diff
@@ -37,7 +37,7 @@ bool CMahalanobisDistance::init(CFeatures* l, CFeatures* r)
 {
 	// FIXME: See comments in
 	// https://github.com/shogun-toolbox/shogun/pull/4085#discussion_r166254024
-	ASSERT(CRealDistance::init(l, r));
```

Comment:
With ASSERT being removed on Release build, there should be a refactor to replace the user facing ones with require.

Reply:
That’s good!
So just to make sure o understand: assert then will become something like in Java: it is ignored unless explicitly asked not to (ie debug mode build)?
That’s good as we can be more liberal with expensive checks.

Reply:
yup